### PR TITLE
2385-feat(front): icon button styles added

### LINKED ISSUE #2385: Correct the "+" icon button style on tables
# Current behavior

CleanShot 2023-11-07 at 15 34 18

The color looks like secondary and the hover take the whole cell.

# Desired behavior

The icon color should be **Tertiary**

its hover should take the full width (fill the red square below)

CleanShot 2023-11-07 at 15 37 29

### PR DESCRIPTION
**Pull request by Kanav-Arora**
Closes #2385 

It required change in standard sized of IconButton component so style property is added to override the IconButton styles.

https://github.com/twentyhq/twenty/assets/62852363/010a52d2-0d0b-4739-a997-78fb5cb39d3f

### CODE REVIEW

File: front/src/modules/ui/object/record-table/components/RecordTableHeader.tsx [hunk 1]
```diff
@@ -210,6 +213,11 @@ export const RecordTableHeader = () => {
                       variant="tertiary"
                       Icon={IconPlus}
                       position="middle"
+                      style={{
```
Comment (charlesBochet) [inline front/src/modules/ui/object/record-table/components/RecordTableHeader.tsx:216]: @Kanav-Arora, I'm not a fan of this approach as this is introducing a way of overwriting the style of our basic components which is not something we want for now as we want to enforce consistency and to be sure that we are not breaking components when updating their API ; this style API is too flexible.

I think here we should actually not use an IconButton. Could you instead :
- add the hover on all RecordTableHeader
- add a onClick on the + Cell itself
- remove this IconButton to use the Icon itself direclty?

Comment (charlesBochet) [reply front/src/modules/ui/object/record-table/components/RecordTableHeader.tsx:216]: Also we should make sure to check the behavior when the columns are taking the whole page width

**Review by charlesBochet — COMMENTED** [review]
Thank you! 
Left comments!

File: front/src/modules/ui/object/record-table/components/RecordTableHeader.tsx (hunk 1 above)
Comment (Kanav-Arora) [reply front/src/modules/ui/object/record-table/components/RecordTableHeader.tsx:216]: Okay I'll proceed with it

**Comment by Kanav-Arora** [comment]
Hi @charlesBochet 

I have done the requested changes since we are not using IconButton component so I have tried to match the sizes and padding according to previous design. 
Moreover I have added the hover effect on other headers too.
Please take a look

https://github.com/twentyhq/twenty/assets/62852363/32272a39-5c1f-4a33-bce2-92f2857c866b

**Comment by charlesBochet** [comment]
@Kanav-Arora taking a look :)

**Review by charlesBochet — APPROVED** [review]
LGTM :)

**Comment by Kanav-Arora** [comment]
Hi @charlesBochet 
In your last commit i.e [Fix alignment](https://github.com/twentyhq/twenty/pull/2411/commits/67ba4664aa49d34814581518de441ff03805c83a), due to removal of <th> the last column is not being rendered as a separate column.

Screenshot 2023-11-09 at 11 19 46 PM

**Comment by charlesBochet** [comment]
Ouch!

**Comment by charlesBochet** [comment]
Good catch, fixed here:
https://github.com/twentyhq/twenty/pull/2417

**Pull request merged**